### ISSUE TITLE
When shall clearing gradients happen?

### ISSUE THREAD

**Issue by sylviachency** [DhavalTaunk08/NLP_scripts#5]
Thank you. In the Transformers multilabel distillbert, in the function "train(epoch)", you clear gradients after the model feed-forward operation and before loss backward calculation, and you uses "optimizer.zero_grad()" twice. Will this clear the gradients that the backward calculation needs? Shall it be like this where we only need to clear gradients once before model feed-forward operation:

        optimizer.zero_grad()
        outputs = model(ids, mask, token_type_ids)

        loss = loss_fn(outputs, targets)
        if _%5000==0:
            print(f'Epoch: {epoch}, Loss:  {loss.item()}')

        loss.backward()
        optimizer.step()

**Comment by DhavalTaunk08** [comment]
Yes, you are right, that was a typo in the code. Corrected it. Thanks,
